### ISSUE TITLE
Element query executed before Craft is fully initialized

### ISSUE THREAD

**Issue by bartdigitalpulse** [solspace/craft3-freeform#1019]
**Describe the bug or issue you're experiencing**

Another "Element query executed before Craft is fully initialized error" in the logs. 
Upgraded a project from Craft 3 to 4.

https://github.com/solspace/craft-freeform/blob/v5/packages/plugin/src/Bundles/Persistence/FormPersistence.php#L24

```
[web.WARNING] [craft\elements\db\ElementQuery::prepare] Element query executed before Craft is fully initialized.

Stack trace:
#0 /var/www/html/vendor/yiisoft/yii2/db/QueryBuilder.php(227): craft\elements\db\ElementQuery->prepare()
#1 /var/www/html/vendor/yiisoft/yii2/db/Query.php(157): yii\db\QueryBuilder->build()
#2 /var/www/html/vendor/yiisoft/yii2/db/Query.php(287): yii\db\Query->createCommand()
#3 /var/www/html/vendor/craftcms/cms/src/db/Query.php(275): yii\db\Query->one()
#4 /var/www/html/vendor/craftcms/cms/src/elements/db/ElementQuery.php(1613): craft\db\Query->one()
#5 /var/www/html/vendor/craftcms/cms/src/elements/User.php(546): craft\elements\db\ElementQuery->one()
#6 /var/www/html/vendor/yiisoft/yii2/web/User.php(698): craft\elements\User::findIdentity()
#7 /var/www/html/vendor/craftcms/cms/src/web/User.php(500): yii\web\User->renewAuthStatus()
#8 /var/www/html/vendor/yiisoft/yii2/web/User.php(199): craft\web\User->renewAuthStatus()
#9 /var/www/html/vendor/solspace/craft-freeform/packages/plugin/src/Bundles/Persistence/FormPersistence.php(24): yii\web\User->getIdentity()
#10 Solspace\Freeform\Bundles\Persistence\FormPersistence->__construct()
#11 /var/www/html/vendor/yiisoft/yii2/di/Container.php(411): ReflectionClass->newInstanceArgs()
#12 /var/www/html/vendor/yiisoft/yii2/di/Container.php(186): yii\di\Container->build()
#13 /var/www/html/vendor/solspace/craft-freeform/packages/plugin/src/Library/Bundles/BundleLoader.php(50): yii\di\Container->get()
#14 /var/www/html/vendor/solspace/craft-freeform/packages/plugin/src/Freeform.php(744): Solspace\Freeform\Library\Bundles\BundleLoader::loadBundles()
#15 /var/www/html/vendor/solspace/craft-freeform/packages/plugin/src/Freeform.php(235): Solspace\Freeform\Freeform->initBundles()
#16 /var/www/html/vendor/yiisoft/yii2/base/BaseObject.php(109): Solspace\Freeform\Freeform->init()
#17 /var/www/html/vendor/yiisoft/yii2/base/Module.php(161): yii\base\BaseObject->__construct()
#18 /var/www/html/vendor/craftcms/cms/src/base/Plugin.php(122): yii\base\Module->__construct()
#19 craft\base\Plugin->__construct()
#20 /var/www/html/vendor/yiisoft/yii2/di/Container.php(419): ReflectionClass->newInstanceArgs()
#21 /var/www/html/vendor/yiisoft/yii2/di/Container.php(170): yii\di\Container->build()
#22 /var/www/html/vendor/yiisoft/yii2/BaseYii.php(365): yii\di\Container->get()
#23 /var/www/html/vendor/craftcms/cms/src/Craft.php(59): yii\BaseYii::createObject()
#24 /var/www/html/vendor/craftcms/cms/src/services/Plugins.php(943): Craft::createObject()
#25 /var/www/html/vendor/craftcms/cms/src/services/Plugins.php(228): craft\services\Plugins->createPlugin()
#26 /var/www/html/vendor/craftcms/cms/src/base/ApplicationTrait.php(1640): craft\services\Plugins->loadPlugins()
#27 /var/www/html/vendor/craftcms/cms/src/web/Application.php(109): craft\web\Application->_postInit()
#28 /var/www/html/vendor/yiisoft/yii2/base/BaseObject.php(109): craft\web\Application->init()
#29 /var/www/html/vendor/yiisoft/yii2/base/Application.php(204): yii\base\BaseObject->__construct()
#30 yii\base\Application->__construct()
#31 /var/www/html/vendor/yiisoft/yii2/di/Container.php(419): ReflectionClass->newInstanceArgs()
#32 /var/www/html/vendor/yiisoft/yii2/di/Container.php(170): yii\di\Container->build()
#33 /var/www/html/vendor/yiisoft/yii2/BaseYii.php(365): yii\di\Container->get()
#34 /var/www/html/vendor/craftcms/cms/src/Craft.php(59): yii\BaseYii::createObject()
#35 /var/www/html/vendor/craftcms/cms/bootstrap/bootstrap.php(250): Craft::createObject()
#36 /var/www/html/vendor/craftcms/cms/bootstrap/web.php(40): require()
#37 /var/www/html/www/index.php(11): require()
```

**Steps to reproduce**

1. On any page load

**Craft & Plugin Info (please complete the following information):**

- Craft Version: Craft Pro 4.7.1
- Freeform Version: 5.0.6.1
- Freeform Edition: Pro
- Fresh Install or Upgrade: Upgrade Craft 3 to 4

**Comment by jrrdnx** [comment]
I've been doing some troubleshooting on this as well, as we're using Sentry to log errors and currently have tens of thousands of these. However, all of ours are coming from `SpamControl/Checks/AbstractCheck.php`.

```
Stack trace:
#0 /path/to/project/craft/vendor/yiisoft/yii2/db/QueryBuilder.php(227): craft\elements\db\ElementQuery->prepare()
#1 /path/to/project/craft/vendor/yiisoft/yii2/db/Query.php(157): yii\db\QueryBuilder->build()
#2 /path/to/project/craft/vendor/yiisoft/yii2/db/Query.php(287): yii\db\Query->createCommand()
#3 /path/to/project/craft/vendor/craftcms/cms/src/db/Query.php(275): yii\db\Query->one()
#4 /path/to/project/craft/vendor/craftcms/cms/src/elements/db/ElementQuery.php(1613): craft\db\Query->one()
#5 /path/to/project/craft/vendor/craftcms/cms/src/elements/User.php(546): craft\elements\db\ElementQuery->one()
#6 /path/to/project/craft/vendor/yiisoft/yii2/web/User.php(698): craft\elements\User::findIdentity()
#7 /path/to/project/craft/vendor/craftcms/cms/src/web/User.php(500): yii\web\User->renewAuthStatus()
#8 /path/to/project/craft/vendor/yiisoft/yii2/web/User.php(199): craft\web\User->renewAuthStatus()
#9 /path/to/project/craft/vendor/yiisoft/yii2/web/User.php(377): yii\web\User->getIdentity()
#10 /path/to/project/craft/vendor/yiisoft/yii2/base/Component.php(139): yii\web\User->getId()
#11 /path/to/project/craft/vendor/solspace/craft-freeform/packages/plugin/src/Bundles/Form/SpamControl/Checks/AbstractCheck.php(26): yii\base\Component->__get()
#12 Solspace\Freeform\Bundles\Form\SpamControl\Checks\AbstractCheck->__construct()
#13 /path/to/project/craft/vendor/solspace/craft-freeform/packages/plugin/src/Freeform.php(978): ReflectionClass->newInstance()
#14 /path/to/project/craft/vendor/solspace/craft-freeform/packages/plugin/src/Freeform.php(303): Solspace\Freeform\Freeform->initBundles()
#15 /path/to/project/craft/vendor/yiisoft/yii2/base/BaseObject.php(109): Solspace\Freeform\Freeform->init()
#16 /path/to/project/craft/vendor/yiisoft/yii2/base/Module.php(161): yii\base\BaseObject->__construct()
#17 /path/to/project/craft/vendor/craftcms/cms/src/base/Plugin.php(122): yii\base\Module->__construct()
#18 craft\base\Plugin->__construct()
#19 /path/to/project/craft/vendor/yiisoft/yii2/di/Container.php(419): ReflectionClass->newInstanceArgs()
#20 /path/to/project/craft/vendor/yiisoft/yii2/di/Container.php(170): yii\di\Container->build()
#21 /path/to/project/craft/vendor/yiisoft/yii2/BaseYii.php(365): yii\di\Container->get()
#22 /path/to/project/craft/vendor/craftcms/cms/src/Craft.php(59): yii\BaseYii::createObject()
#23 /path/to/project/craft/vendor/craftcms/cms/src/services/Plugins.php(943): Craft::createObject()
#24 /path/to/project/craft/vendor/craftcms/cms/src/services/Plugins.php(228): craft\services\Plugins->createPlugin()
#25 /path/to/project/craft/vendor/craftcms/cms/src/base/ApplicationTrait.php(1640): craft\services\Plugins->loadPlugins()
#26 /path/to/project/craft/vendor/craftcms/cms/src/web/Application.php(109): craft\web\Application->_postInit()
#27 /path/to/project/craft/vendor/yiisoft/yii2/base/BaseObject.php(109): craft\web\Application->init()
#28 /path/to/project/craft/vendor/yiisoft/yii2/base/Application.php(204): yii\base\BaseObject->__construct()
#29 yii\base\Application->__construct()
#30 /path/to/project/craft/vendor/yiisoft/yii2/di/Container.php(419): ReflectionClass->newInstanceArgs()
#31 /path/to/project/craft/vendor/yiisoft/yii2/di/Container.php(170): yii\di\Container->build()
#32 /path/to/project/craft/vendor/yiisoft/yii2/BaseYii.php(365): yii\di\Container->get()
#33 /path/to/project/craft/vendor/craftcms/cms/src/Craft.php(59): yii\BaseYii::createObject()
#34 /path/to/project/craft/vendor/craftcms/cms/bootstrap/bootstrap.php(250): Craft::createObject()
#35 /path/to/project/craft/vendor/craftcms/cms/bootstrap/web.php(40): require()
#36 /path/to/project/craft/web/index.php(14): require()
```

- Craft Versions: Craft Pro 4.5.x to 4.7.x (all fresh installs, no upgrades)
- Freeform Version: 4.x
- Freeform Edition: Pro

**Comment by kjmartens** [comment]
Sorry for the trouble you're experiencing @bartdigitalpulse and @jrrdnx.

We'll investigate this issue shortly.

**Comment by kjmartens** [comment]
This is now resolved in Freeform 5.0.10+. 🙂
